torchaudio/prototype/functional/functional.py

```diff
@@ -0,0 +1,42 @@
+import math
+
+import torch
+
+
+def convolve(x: torch.Tensor, y: torch.Tensor, use_fft: bool = False) -> torch.Tensor:
+    """
+    Convolves inputs along last dimension. Note that, in contrast to ``torch.nn.functional.conv1d``, this
+    function applies the true `convolution`_ operator.
+
+    Args:
+        x (torch.Tensor): First convolution operand, with shape `(*, N)`.
+        y (torch.Tensor): Second convolution operand, with shape `(*, M)`
+            (leading dimensions must match those of ``x``).
+        use_fft (bool, optional): If ``True``, then convolve ``x`` and ``y`` using FFT.
+            Otherwise, convolve ``x`` and ``y`` directly via sums. (Default: ``False``)
+
+    Returns:
+        torch.Tensor: Result of convolving ``x`` and ``y``, with shape `(*, N + M - 1)`, where
+        the leading dimensions match those of ``x``.
+
+    .. _convolution:
+        https://en.wikipedia.org/wiki/Convolution
+    """
+    if use_fft:
+        n = x.size(-1) + y.size(-1) - 1
+        fresult = torch.fft.rfft(x, n=n) * torch.fft.rfft(y, n=n)
+        output = torch.fft.irfft(fresult, n=n)
```

Comment:
When `x` and `y` are both of dtype `int` or `long`, using FFT solution will change the dtype to `float` or `double`. Could you cast the output to be the same dtype as the input?
```suggestion
        output = torch.fft.irfft(fresult, n=n).to(x.dtype)
```

Reply:
i feel like it makes more sense to keep the output dtype as float and communicate that to users (scipy does this)

Reply:
I think automatically converting float into int is dangerous. It will lose many precision. It can be performed in client code if necessary.

Reply:
i see, sounds good